### ISSUE TITLE
Use $HOME or ~ in $HOME/.zim/init.zsh

### ISSUE THREAD

**Issue by markfaine** [zimfw/zimfw#509]
**Is your feature request related to a problem? Please describe.**  

I sometimes load a docker container with my home directory mounted.  In the container, it's a different user, though mapped to my UID, however, because `/home/username` is hard-coded into `init.zsh` , it results in a bunch of errors when zsh loads zim in the container.
I've worked around it by replace `/home/username` with `$HOME`, however, I have to do this anytime `zimfw` runs because it reverts my changes.

**Describe the solution you'd like**  
Replace the path to the user's home directory with `$HOME` in `init.zsh`

**Describe alternatives you've considered**  
I guess since it's not quoted `~` works just as well, though I think `$HOME` is probably a better choice.

**Comment by ericbn** [comment]
Hi. I've changed the code in the [use-home-in-init-zsh](https://github.com/zimfw/zimfw/tree/use-home-in-init-zsh) branch to generate the init.zsh file with `${HOME}` instead of the resolved path. I've run our benchmarks in my local Intel MacBook Pro and observed a 19% increase in the startup time with this new change:
```
framework,mean,stddev,min,max
zim,0.0850283,0.00218265,0.082452,0.089822
zim-branch1,0.101417,0.00560655,0.095258,0.113082
```

Maybe having the resolved paths is a performance optimization worth keeping. Will try to run the same benchmarks with GitHub Actions to have those measurements to assess too.

**Comment by ericbn** [comment]
Didn't have the change to come back to this, but I tend to agree that it's better to keep the `${HOME}` variable unresolved in the init.zsh so it's more friendly to edge case scenarios, than try to be minimally faster. Just want to try to run the benchmark in the GitHub Actions machine once to see how it affects that benchmark. Any help setting that up is welcome.

@markfaine, to download the zimfw.zsh from the mentioned branch, use https://raw.githubusercontent.com/zimfw/zimfw/use-home-in-init-zsh/zimfw.zsh. That branch is now way behind the master branch, and it's on top of zimfw v1.11.3.

**Comment by ericbn** [comment]
The results of the benchmark in the GitHub Actions machine is in: https://github.com/zimfw/zsh-framework-benchmark/actions/runs/9291824609/job/25571266253

framework | mean | stddev | min | max
-- | -- | -- | -- | --
vanilla | 0.00529283 | 0.00019565 | 0.005089 | 0.006581
zinit-turbo | 0.0448112 | 0.00116446 | 0.04324 | 0.051293
**zim-branch1 (use-home-in-init-zsh)** | 0.0573027 | 0.0014278 | 0.055373 | 0.063893
**zim (master)** | 0.0573594 | 0.00181664 | 0.055303 | 0.066643
zcomet | 0.0663811 | 0.00214085 | 0.06476 | 0.084694
antibody-static | 0.0691344 | 0.00067308 | 0.068107 | 0.072547
zgen | 0.0723409 | 0.00175374 | 0.070652 | 0.081492
antigen | 0.0804912 | 0.00058769 | 0.079479 | 0.082419
prezto | 0.097302 | 0.00206302 | 0.095081 | 0.111511
ohmyzsh | 0.136848 | 0.00101534 | 0.135207 | 0.140645
zplug | 0.216071 | 0.00225839 | 0.211897 | 0.235223
zinit-light | 0.228825 | 0.00305681 | 0.222101 | 0.238459


Looking good, so this is something worth merging to master and releasing.

**Comment by ericbn** [comment]
@markfaine, I just realized this conflicts with #486. How are we going to allow a machine to have just one init.zsh script that can be sourced by different users if the script uses `${HOME}` which varies between users?

**Comment by ericbn** [comment]
Fair enough, I agree this is a good convention:
* If ${ZIM_HOME} is under ${HOME}, then generate init.zsh with ${HOME} unexpanded, as request in this issue. This assumes the init.zsh won't be shared with other users.
* Otherwise, generate init.zsh with fixed paths (which will happen automatically, since there won't be any paths under ${HOME}). This generates an init.zsh file that can be shared with other users, as requested in #486.

**Comment by ericbn** [comment]
Just released zimfw v1.14.0 with this fix.
